Explorer/Assets/DCL/Profiles/ProfileJsonDto.cs

```diff
@@ -0,0 +1,205 @@
+using CommunicationData.URLHelpers;
+using DCL.Optimization.ThreadSafePool;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace DCL.Profiles
+{
+    [Serializable]
+    public class EmoteJsonDto
+    {
+        public int slot;
+        public string urn;
+
+        public Emote ToEmote() =>
+            new (slot, urn);
+    }
+
+    [Serializable]
+    public class AvatarColorJsonDto
+    {
+        public float r;
+        public float g;
+        public float b;
+        public float a;
+
+        public Color ToColor() =>
+            new (r, g, b, a);
+
+        public void Reset()
+        {
+            r = 1;
+            g = 1;
+            b = 1;
+            a = 1;
+        }
+    }
+
+    [Serializable]
+    public class EyesJsonDto
+    {
+        public AvatarColorJsonDto? color;
+    }
+
+    [Serializable]
+    public class HairJsonDto
+    {
+        public AvatarColorJsonDto? color;
+    }
+
+    [Serializable]
+    public class SkinJsonDto
+    {
+        public AvatarColorJsonDto? color;
+    }
+
+    [Serializable]
+    public class AvatarSnapshotJsonDto
+    {
+        public string face256;
+        public string body;
+    }
+
+    [Serializable]
+    public class AvatarJsonDto
+    {
+        public string? bodyShape;
+        public List<string>? wearables;
+        public List<string>? forceRender;
+        public List<EmoteJsonDto>? emotes;
+        public AvatarSnapshotJsonDto? snapshots;
+        public EyesJsonDto? eyes;
+        public HairJsonDto? hair;
+        public SkinJsonDto? skin;
+
+        public void CopyTo(Avatar avatar)
+        {
+            const int SHARED_WEARABLES_MAX_URN_PARTS = 6;
+
+            var sharedWearables = new HashSet<string>(wearables.Count);
+
+            foreach (string wearable in wearables)
+                sharedWearables.Add(wearable.ShortenURN(SHARED_WEARABLES_MAX_URN_PARTS));
+
+            // To avoid inconsistencies in the wearable references thus improving cache miss rate,
+            // we keep a list of shared wearables used by avatar shapes and most of the rendering systems
+            avatar.SharedWearables = sharedWearables;
+            // The wearables urns retrieved in the profile follows https://adr.decentraland.org/adr/ADR-244
+            avatar.UniqueWearables = new HashSet<string>(wearables);
```

Comment:
We could save extra allocations by clearing->filling lists, but they are readonly. Any workaround?

Reply:
I modified this part to expose non-readonly collections as `internal` and manipulate them instead